wavefront/server/apps/inference_app/inference_app/service/image_embedding.py

```
@@ -69,3 +69,49 @@ def query_embed(self, image_content: bytes) -> List[Dict[str, List[float]]]:
+          ]
+        """
+        if not image_batch:
+            return []
+
+        # Decode bytes -> PIL images on CPU.
+        # The actual model forward pass (processor->tensor + model) is batched on GPU.
+        images: List[Image.Image] = []
+        for idx, image_content in enumerate(image_batch):
+            try:
+                images.append(Image.open(io.BytesIO(image_content)).convert('RGB'))
+            except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)